CRM/Utils/Weight.php

```diff
@@ -154,12 +151,18 @@ public static function updateOtherWeights($daoName, $oldWeight, $newWeight, $fie
       return $newWeight;
     }
 
+    // Check for an existing record with this weight
+    $existing = self::query('SELECT', $daoName, $fieldValues, "id", "$weightField = $newWeight");
+    // Nothing to do if no existing record has this weight
+    if (empty($existing->N)) {
+      return $newWeight;
```

Comment:
I'm not 100% sure about this because there's a couple places where the comments indicate it's supposed to create a gap for the new row, and there are several calls to this function. But logically the gap would already be there if it doesn't exist. I tried clicking around a couple places and it seems ok.

I'm not sure how to evaluate the api/search kit changes since I'm not sure where they're used. So have given merge-ready.

Reply:
@demeritcowboy I think it was an oversight. Prior to this PR there were no unit tests for `CRM_Utils_Weight`.